### ISSUE TITLE
Log::Report::Dispatcher::File loops forever on failure

### ISSUE THREAD

**Issue by tomhukins** [markov2/perl5-Log-Report#4]
When the file dispatcher is configured to write to a file it lacks write permissions for, it hits an infinite loop:

```perl
#!perl

use strict;
use warnings;

use File::Temp 'tempfile';
use Log::Report mode => 'DEBUG';

my ($fh, $filename) = tempfile();
chmod 0, $filename;

dispatcher FILE => 'b0rked',
    to => $filename;

info 'Whee!';
```

This outputs errors repeatedly until I terminate the process:

```
fault: cannot write log into /tmp/AXhESIvkLk with mode >>; Permission denied
 at break_log_report.pl line 15
Log::Report::info("Whee!") at break_log_report.pl line 15
```

**Comment by markov2** [comment]
Thanks for your report Tom.  (I hope you are doing well!)  The cause is quite obvious: the tempfile is opened lazily... when it gets opened to log the info, the error catching dispatcher to the FILE is already active.  Let's see what I can do about that.

In this case, I would pass the $fh as 'to': otherwise we reopen the tempfile needlessly.

**Comment by tomhukins** [comment]
Hi, Mark - thanks for your quick reply.  Fortunately I have an easy workaround of making log files writeable, but I hope you find a way to handle the problem more gracefully in Log::Report.

**Comment by markov2** [comment]
Will be fixed in 1.30
